src/charm.py

```diff
@@ -401,6 +412,36 @@ def _update_config(self) -> None:
         self._push_config_and_reload(pending)
         self._stored.config_hash = config_hash
 
+    def _get_config(self) -> Union[dict, None]:
+        local_config = self.config["config_file"]
+        remote_config = self.remote_configurer_provider.config()
+        if local_config and remote_config:
+            logger.error("unable to use config from config_file and relation at the same time")
+            raise ConfigUpdateFailure("Multiple configs detected")
```

Comment:
Is this really true? Doesn't alertmanager merge these automatically? If we've decided to make the options exclusive, I'm totally fine with that, but people might want to use the same alertmanager with both Telco and other charms.

Reply:
My vote is definitely on having a single source of configuration. Merging multiple configs may not be a problem with `routes` or `receivers`, but what if different configs provide mutually exclusive global configs (like TLS for instance)?